evmcore/state_processor.go

```diff
@@ -177,6 +177,15 @@ func applyTransaction(
 	// Skip checking of base fee limits for internal transactions.
 	evm.Config.NoBaseFee = msg.SkipAccountChecks
 
+	// For now, Sonic only supports Blob transactions without blob data.
+	if msg.BlobHashes != nil {
```

Comment:
I think at this point it is to late to filter out the transactions we can not accept. At this point, the list of transactions to be included in a block is already fixed, and with #278 properties will already have been derived from this list of transactions (in particular the order of the transactions).

I think the filtering should happen [here](https://github.com/Fantom-foundation/Sonic/blob/a75bcb23ff6345f22dbbb53fe61382d583354c36/gossip/c_block_callbacks.go#L311-L313), before the list of transactions of a block is passed to the processor. Also at this place, the transaction re-ordering is going to be added.

Reply:
Yes, this is what we once discussed. Nevertheless, I discovered that it does exist the possibility that the processor skips the transaction (what was happening already with the ones with non-nil but empty blobhashes).

questions: 
- shall it continue to be ignored or do we want to craft a failure receipt? 
- the patching of empty->nil blobhases still needs to be done at the current place. Unless we want to reject transactions with non-nil but empty bloblhases list.

Reply:
You are right, we do already skip other transactions as well. So my original thought of being able to verify the scrambling is anyway already broken -- and as I just realized, also not necessary. So yes, in this case we can leave it as you suggested it.